### PR TITLE
NDRS-1030: Disconnect from faulty peers

### PR DESCRIPTION
**Pull request by marc-casperlabs**
This PR creates a new announcement for blocklisting, which is currently picked up by `small_network`. If enabled via configuration (currently disabled by default), the peer will be blocked. Regardless of configuration the event is logged.

As a consequence, it is possible to punish peers not only from consensus. As a nice side effect, the `ConsensusAnnouncement` is no longer generic over `I`.

### CODE REVIEW

File: node/src/reactor/validator.rs [hunk 1]
```diff
@@ -185,6 +186,9 @@ pub enum Event {
     /// Chainspec loader announcement.
     #[from]
     ChainspecLoaderAnnouncement(#[serde(skip_serializing)] ChainspecLoaderAnnouncement),
+    /// Blocklist announcement.
+    #[from]
+    BlocklistAnnouncement(BlocklistAnnouncement<NodeId>),
```
Comment (afck) [inline node/src/reactor/validator.rs:191]: Most variants `#[serde(skip_serializing)]` — not sure which ones should and shouldn't?

**Review by afck — APPROVED** [review]
Looks good to me! :+1: 
It's not actually a blacklist/"blocklist" (yet?), though — it just disconnects but would allow the peer to reconnect immediately, would it?

File: node/src/reactor/validator.rs (hunk 1 above)
Comment (marc-casperlabs) [reply node/src/reactor/validator.rs:191]: `skip_serializing` is used for values where there is no `Serialize` implementation. Some might not have been updated.

**Comment by marc-casperlabs** [comment]
> It's not actually a blacklist/"blocklist" (yet?), though — it just disconnects but would allow the peer to reconnect immediately, would it?

There's a parameter that decides blocking on `fn remove()`, the last one.

**Review by xcthulhu — APPROVED** [review]

**Review by Fraser999 — APPROVED** [review]
LGTM, but shouldn't we wire it up for the joiner reactor too?  And if so, should we try and persist the blocklist from joiner to validator instances of small_network?

**Pull request merged**